### PR TITLE
Add -AsJob parameter to websites cmdlets

### PR DESCRIPTION
**Pull request by panchagnula**
## Description

---
https://github.com/Azure/azure-powershell/issues/5120
This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1 [hunk 1]
```diff
@@ -621,7 +625,7 @@ function Test-RemoveWebAppSlot
 		Assert-AreEqual $serverFarm.Id $slot.ServerFarmId
 
 		# Remove web app via pipeline obj
-		$slot | Remove-AzureRmWebAppSlot -Force
+		$slot | Remove-AzureRmWebAppSlot -Force -AsJob
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: Need to add Wait-Job here

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1 [hunk 2]
```diff
@@ -43,7 +43,7 @@ function Test-GetWebAppSlot
 		Assert-AreEqual $serverFarm.Id $webapp.ServerFarmId
 		
 		# Create new deployment slot
-		$slot1 = New-AzureRmWebAppSlot -ResourceGroupName $rgname -Name $appname -Slot $slotname1 -AppServicePlan $planName 
+		$slot1 = New-AzureRmWebAppSlot -ResourceGroupName $rgname -Name $appname -Slot $slotname1 -AppServicePlan $planName
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: Looks like you didn't add the test here.

File: src/ResourceManager/Websites/Commands.Websites/help/Remove-AzureRmWebApp.md [hunk 3]
```diff
@@ -15,13 +15,13 @@ Removes an Azure Web App.
 
 ### S1
 ```
-Remove-AzureRmWebApp [-Force] [-ResourceGroupName] <String> [-Name] <String>
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites/help/Remove-AzureRmWebApp.md]: There doesn't seem to be a test for this cmdlet.

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1 (hunk 2 above)
Comment (panchagnula) [reply src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: This is the Test-GetWebAppSlot , i haven't updated this one i add it to the Test-CreateNewWebAppSlot

File: src/ResourceManager/Websites/Commands.Websites/help/Remove-AzureRmWebApp.md (hunk 3 above)
Comment (panchagnula) [reply src/ResourceManager/Websites/Commands.Websites/help/Remove-AzureRmWebApp.md]: WebAppTests.ps1 has this "function Test-RemoveWebApp" which is updated. Is there anything else i am missing here? Thanks!

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1 (hunk 2 above)
Comment (maddieclayton) [reply src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: So in Test-CreateNewWebAppSlot, I see you updated the New-AzureRmWebApp cmdlet but not the New-AzureRmWebAppSlot cmdlet, so at the moment there is no test for New-AzureRmWebAppSlot.

Comment (panchagnula) [reply src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: Thanks for pointing this out - updated now

**Comment by maddieclayton** [comment]
On demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/340/

**Pull request merged**